### PR TITLE
Tidy up tests

### PR DESCRIPTION
**Pull request by OiYouYeahYou**
## Description

* Add nyc coverage tool
* Remove `env-test`: it's a single line for the cost of a module
* Set NODE_ENV in custom set up file
* Stub console for each test individually globally
* Plus: a `sinon.restore()` in global afterEach prevents cross test leaks
* Add most basic test for `Tldr` class, just to trigger coverage
* Remove surplus mocha args:
        * It always looks in test dir
        * `mocha.opts` provides a more consistent approach
* Ignore coverage files now created by nyc

## Checklist

Please review this checklist before submitting a pull request.

- [x] Code compiles correctly
- [x] Created tests, if possible
- [x] All tests passing (`npm run test:all`)
- [ ] Extended the README / documentation, if necessary

### CODE REVIEW

File: test/mocha.js [hunk 1]
```diff
@@ -0,0 +1,11 @@
+process.env.NODE_ENV = 'test';
```
Comment (vladimyr) [inline test/mocha.js:1]: Move this to `mocha.opts` as `--env` flag cause that is where people would typically look for it. Add `use strict` pragma on the top of the file.

Comment (OiYouYeahYou) [reply test/mocha.js:1]: Could you give me the syntax of a `--env` flag with mocha? I'm looking for it's documentation and have thrown some stuff at the wall, but nothing seems to work.

File: test/mocha.opts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 --reporter spec
```
Comment (agnivade) [inline test/mocha.opts:1]: While we are here, can we move from mocha.opts to a config file ? Looks like it is deprecated and will be removed in future: https://mochajs.org/#mochaopts. `.mocharc.js` looks fine by me.

File: test/mocha.js (hunk 1 above)
Comment (agnivade) [reply test/mocha.js:3]: I don't think there _is_ an `--env` flag: https://github.com/mochajs/mocha/issues/185. I am good with this or setting it via a variable in the npm invocation string.

**Pull request merged**